Design/Workbench/design-layout.md

```diff
@@ -0,0 +1,149 @@
+       |       |-- widgets # would contain combine qtwidgetscommon.dll with sip exports to give mantidqt.widgets python library
+       |       |-- __init__.py
+       |-- widgets
+           |-- common # qtwidgetscommon.dll
+           |-- plugins
+           |   |-- algorithm_dialogs # qtwidgetspluginsalgorithm_dialogs.dll
+           |   |-- designer # qtwidgetspluginsdesigner.dll
+           |-- instrumentview # qtwidgetsinstrumentview.dll
+           |-- spectrumviewer # qtwidgetsspectrumviewer.dll
+           |-- sliceviewer # qtwidgetssliceviewer.dll
+```
+
+Mapping of the current structure to this is is as follows:
+
+ - `/MantidPlot`: NOT MOVED (will be deleted once replacement is up to scratch)
+ - `/Vates` --> `/qt/paraview_ext`
+ - `/MantidQt/API/` & `/MantidQt/MantidWidgets/` combined --> `/qt/widgets/common`
+ - `/MantidQt/MantidWidgets/InstrumentView` --> `qt/widgets/instrumentview`
+ - `/MantidQt/RefDetectorViewer` --> `/qt/widgets/refdetectorview`
+ - `/MantidQt/SpectrumViewer` --> `/qt/widgets/spectrumviewer`
+ - `/MantidQt/SliceViewer` --> `/qt/widgets/sliceviewer`
+ - `/MantidQt/CustomDialogs` --> `/qt/widgets/plugins/algorithm_dialogs`
+ - `/MantidQt/DesignerPlugins` --> `/qt/widgets/plugins/designer`
+ - `/MantidQt/Python` --> `/qt/python`
+ - Remove Factory directory as it seems unecessary
+ - All resources, fonts, images etc, moved under qt/resources
+
+Shared libraries follow a naming structure that includes all of their parent directories.
+
+Technologies
+============
+
+The workbench will be written primarily in Python using:
+
+ - [PyQt5][PyQt5]
+ - [matplotlib][matplotlib_org]
+ - [IPython][IPython]
+
+`sip` will be used for exporting any required C/C++ to Python.
+
+Python 2/3
+----------
+
+Due to the requirement to support RedHat 7 we will continue to write Python 2/3 compatible
+code.
+
+PyQt shim
+----------
+
+An abstraction layer around `PyQt4`, `PyQt5`, `PySide` will be used to wrap all calls to PyQt functionality. This should reduce
+portability concerns in the future when its inevitable that a PyQt6 will come out. The shim should provide a similar layer
+of protection as it currently does between `PyQt4` & `PyQt5`.
+
+We could either write our own, use those provided by other dependencies: [matplotlib][matplotlib_qtcompat], [IPython][IPython] or
+use a separate package such as [qtpy][qtpy] provided by the [Spyder][Spyder] developers.
+
+Packaging & Deployment
+======================
+
+Python
+------
+
+The framework package will remain separate and called `mantid`.
+
+The new ui package will be called `mantidui` and have the following submodules:
+
+ - mantidui.plotting: contain custom plotting code based on matplotlib, i.e keep/make current behaviour, custom toolbars, custom figure window
+ - mantidui.widgets: contain these set of reusable widgets used to build the workbench & its components
+
+The workbench will be called `mantidplot` and depend on `mantudui` & `mantid`.
```

Comment:
Do we really want to call it MantidPlot?  That surely would cause confusion.  I'd go with MantidBench or Mantid Workbench for now.